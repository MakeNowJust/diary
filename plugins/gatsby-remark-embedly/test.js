import {remark} from 'remark';

import embedly from './main';

test('add embedly-card class to link which is only child of paragraph', () => {
  const context = {markdownAST: remark.parse('<http://example.com/>')};
  embedly(context);
  expect(context.markdownAST).toMatchSnapshot('link');
});

test('dont add embedly-card class to non link which is only child of paragraph', () => {
  const context = {markdownAST: remark.parse('non link')};
  embedly(context);
  expect(context.markdownAST).toMatchSnapshot('non-link');
});
