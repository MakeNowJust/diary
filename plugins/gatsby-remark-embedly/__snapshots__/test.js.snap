// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add embedly-card class to link which is only child of paragraph: link 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 20,
                  "line": 1,
                  "offset": 19,
                },
                "indent": Array [],
                "start": Object {
                  "column": 1,
                  "line": 1,
                  "offset": 0,
                },
              },
              "type": "text",
              "value": "http://example.com/",
            },
          ],
          "data": Object {
            "hProperties": Object {
              "className": "embedly-card",
            },
          },
          "position": Position {
            "end": Object {
              "column": 20,
              "line": 1,
              "offset": 19,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "http://example.com/",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`dont add embedly-card class to non link which is only child of paragraph: non-link 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 9,
              "line": 1,
              "offset": 8,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "text",
          "value": "non link",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
